File: C/Partie7/Exo28affiche.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <signal.h>

int entier = 0;

void handler(int sig)
{
    FILE *f = fopen("/tmp/entier.txt", "r");
    fscanf(f, "%d", &entier);
    fclose(f);
    printf("Entier reçu : %d\n", entier);
}

int main()
{
    signal(SIGUSR1, handler);
    while (1)
    {
        sleep(1);
    }
    return 0;
}

File: Python/Partie7/Exo28affiche.py
import signal, time

entier = 0
def handler(sig, frame):
    global entier
    with open("/tmp/entier.txt", "r") as f:
        entier = int(f.read())
    print(f"Entier reçu : {entier}")

signal.signal(signal.SIGUSR1, handler)
while True:
    time.sleep(1)